diff --git a/clone.sh b/clone.sh
index 786ae43..00f1901 100755
--- a/clone.sh
+++ b/clone.sh
@@ -7,8 +7,8 @@ if [ $VERSION = "stretch" ] || [ $VERSION = "16.04" ]; then
 fi
 
 # Basic dependencies
-echo "✅ Installing dependencies..."
-sudo apt install -y make patch rsync git cmake ninja-build clang$CLANG_VERSION python uuid-dev libicu-dev icu-devtools libbsd-dev libedit-dev libxml2-dev libsqlite3-dev swig libpython-dev libncurses5-dev pkg-config libblocksruntime-dev libcurl4-openssl-dev autoconf libtool systemtap-sdt-dev libcurl4-openssl-dev libz-dev
+#echo "✅ Installing dependencies..."
+#sudo apt install -y make patch rsync git cmake ninja-build clang$CLANG_VERSION python uuid-dev libicu-dev icu-devtools libbsd-dev libedit-dev libxml2-dev libsqlite3-dev swig libpython-dev libncurses5-dev pkg-config libblocksruntime-dev libcurl4-openssl-dev autoconf libtool systemtap-sdt-dev libcurl4-openssl-dev libz-dev
 
 # Some OS or specific versions could have additional requirements, if they do
 # a script will be present in distro-scripts and we'll run it right after the more generic configuration
@@ -23,4 +23,4 @@ done
 # Clone Swift and all the related projects
 git clone https://github.com/apple/swift
 
-./swift/utils/update-checkout --clone
+./swift/utils/update-checkout --clone
diff --git a/utils.sh b/utils.sh
index f4a0e58..b05ad31 100644
--- a/utils.sh
+++ b/utils.sh
@@ -1,37 +1,7 @@
 #!/bin/sh
 
 # Set OS, VERSION and ARCH
-OS="unknown"
-VERSION="unknown"
-ARCH="unknown"
-ARCHFAMILY="unknown"
-
-if [[ $(cat /etc/os-release) = *"Ubuntu"* ]]; then
-    OS="ubuntu"
-    VERSION=`cat /etc/os-release | sed -n 's/VERSION_ID=\"\([0-9].*\)\"/\1/p'`
-elif [[ $(cat /etc/os-release) = *"Raspbian"* ]]; then
-    OS="raspbian"
-    if [[ $(cat /etc/os-release) = *"stretch"* ]]; then
-        VERSION="stretch"
-    elif [[ $(cat /etc/os-release) = *"buster"* ]]; then
-        VERSION="buster"
-    fi
-elif [[ $(cat /etc/os-release) = *"stretch"* ]]; then
-    OS="debian"
-    VERSION="stretch"
-elif [[ $(cat /etc/os-release) = *"buster"* ]]; then
-    OS="debian"
-    VERSION="buster"
-fi
-
-case `uname -m` in
-    aarch64)
-        ARCH="aarch64"
-        ARCHFAMILY="aarch64";;
-    armv6*)
-        ARCH="armv6"
-        ARCHFAMILY="aarch32";;
-    *)
-        ARCH="armv7"
-        ARCHFAMILY="aarch32";;
-esac
+OS="raspbian"
+VERSION="buster"
+ARCH="armv6"
+ARCHFAMILY="aarch32"
